File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsPaths.swift
import Foundation

// MARK: - Learning paths (M8.2)

struct CatalogPathSummary: Codable, Identifiable, Hashable {
    var id: String
    var title: String
    var description: String
    var slug: String
    var bundlePriceCents: Int?
    var courseCount: Int
    var totalDurationMinutes: Int
    var individualTotalCents: Int
    var skillTags: [String]
}

struct CatalogPathsListResponse: Decodable {
    var paths: [CatalogPathSummary]?
}

struct PathCourseProgress: Codable, Identifiable, Hashable {
    var courseId: String
    var position: Int
    var courseCode: String
    var title: String
    var description: String?
    var listPriceCents: Int?
    var durationMinutes: Int?
    var skillTags: [String]?
    var completed: Bool?
    var recommended: Bool?

    var id: String { courseId }

    var isCompleted: Bool { completed == true }
    var isRecommended: Bool { recommended == true }
}

struct PathProgress: Codable, Identifiable, Hashable {
    var pathId: String
    var pathTitle: String
    var slug: String?
    var totalCourses: Int
    var completedCourses: Int
    var percent: Int
    var progressLabel: String
    var completedAt: String?
    var justCompleted: Bool
    var courses: [PathCourseProgress]

    var id: String { pathId }
}

struct MyPathsListResponse: Decodable {
    var paths: [PathProgress]?
}

struct LearningPathDetailPath: Decodable {
    var id: String
    var title: String
    var description: String
    var slug: String?
    var bundlePriceCents: Int?
    var isPublic: Bool
}

struct LearningPathDetail: Decodable {
    var path: LearningPathDetailPath
    var courses: [PathCourseProgress]
    var totalDurationMinutes: Int
    var individualTotalCents: Int
    var skillTags: [String]
    var slug: String
}

struct PathEnrollResponse: Decodable {
    var enrollmentId: String
    var progress: PathProgress?
}

struct RecommendationEventBody: Encodable {
    var courseId: String
    var itemId: String?
    var surface: String
    var eventType: String
    var rank: Int?
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsPaths.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class CatalogPathSummary(
    val id: String,
    val title: String,
    val description: String = "",
    val slug: String,
    val bundlePriceCents: Int? = null,
    val courseCount: Int = 0,
    val totalDurationMinutes: Int = 0,
    val individualTotalCents: Int = 0,
    val skillTags: List<String> = emptyList(),
)

@Serializable
data class CatalogPathsListResponse(
    val paths: List<CatalogPathSummary> = emptyList(),
)

@Serializable
data class PathCourseProgress(
    val courseId: String,
    val position: Int,
    val courseCode: String,
    val title: String,
    val description: String? = null,
    val listPriceCents: Int? = null,
    val durationMinutes: Int? = null,
    val skillTags: List<String>? = null,
    val completed: Boolean? = null,
    val recommended: Boolean? = null,
) {
    val isCompleted: Boolean get() = completed == true
    val isRecommended: Boolean get() = recommended == true
}

@Serializable
data class PathProgress(
    val pathId: String,
    val pathTitle: String,
    val slug: String? = null,
    val totalCourses: Int = 0,
    val completedCourses: Int = 0,
    val percent: Int = 0,
    val progressLabel: String = "",
    val completedAt: String? = null,
    val justCompleted: Boolean = false,
    val courses: List<PathCourseProgress> = emptyList(),
)

@Serializable
data class MyPathsListResponse(
    val paths: List<PathProgress> = emptyList(),
)

@Serializable
data class LearningPathDetailPath(
    val id: String,
    val title: String,
    val description: String = "",
    val slug: String? = null,
    val bundlePriceCents: Int? = null,
    val isPublic: Boolean = true,
)

@Serializable
data class LearningPathDetail(
    val path: LearningPathDetailPath,
    val courses: List<PathCourseProgress> = emptyList(),
    val totalDurationMinutes: Int = 0,
    val individualTotalCents: Int = 0,
    val skillTags: List<String> = emptyList(),
    val slug: String = "",
)

@Serializable
data class PathEnrollResponse(
    val enrollmentId: String,
    val progress: PathProgress? = null,
)

@Serializable
data class RecommendationEventBody(
    val courseId: String,
    val itemId: String? = null,
    val surface: String,
    val eventType: String,
    val rank: Int? = null,
)